addons/library.xbmc.addon/libXBMC_addon.h

```diff
@@ -264,6 +266,14 @@ namespace ADDON
         dlsym(m_libXBMC_addon, "XBMC_remove_directory");
       if (XBMC_remove_directory == NULL) { fprintf(stderr, "Unable to assign function %s\n", dlerror()); return false; }
 
+      XBMC_get_directory = (bool (*)(void* HANDLE, void* CB, const char* strPath, const char* mask, VFSDirEntry** items, int* num_items))
+        dlsym(m_libXBMC_addon, "XBMC_get_directory");
+      if (XBMC_get_directory == NULL) { fprintf(stderr, "Unable to assign function %s\n", dlerror()); return false; }
+
+      XBMC_free_directory = (void (*)(void* HANDLE, void* CB, VFSDirEntry* items, int num_items))
+        dlsym(m_libXBMC_addon, "XBMC_free_directory");
+      if (XBMC_free_directory == NULL) { fprintf(stderr, "Unable to assign function %s\n", dlerror()); return false; }
+
```

Comment:
If the policy to keep the api c has changed it is an option. Should be fairly hassle less with the c++11 abi standardization.

-------- Opprinnelig melding --------
Fra: Alwin Esch <notifications@github.com>
Dato: 27.11.2015 18:03 (GMT+01:00)
Til: xbmc/xbmc <xbmc@noreply.github.com>
Ko: Arne Morten Kvarving <Arne.Morten.Kvarving@sintef.no>
Emne: Re: [xbmc] added: addon callback for listing a directory (#8450)


In addons/library.xbmc.addon/libXBMC_addon.h<https://github.com/xbmc/xbmc/pull/8450#discussion_r46061875>:

> @@ -264,6 +266,14 @@ namespace ADDON
>          dlsym(m_libXBMC_addon, "XBMC_remove_directory");
>        if (XBMC_remove_directory == NULL) { fprintf(stderr, "Unable to assign function %s\n", dlerror()); return false; }
>
> +      XBMC_get_directory = (bool (*)(void* HANDLE, void* CB, const char* strPath, const char* mask, VFSDirEntry** items, int* num_items))
> +        dlsym(m_libXBMC_addon, "XBMC_get_directory");
> +      if (XBMC_get_directory == NULL) { fprintf(stderr, "Unable to assign function %s\n", dlerror()); return false; }
> +
> +      XBMC_free_directory = (void (*)(void* HANDLE, void* CB, VFSDirEntry* items, int num_items))
> +        dlsym(m_libXBMC_addon, "XBMC_free_directory");
> +      if (XBMC_free_directory == NULL) { fprintf(stderr, "Unable to assign function %s\n", dlerror()); return false; }
> +


One point for the library rework is that also dlsym is no more needed, except the "Register" and "Unregister".

There comes also a question (idea). Possible to use then std::vector and std::string direct and prevent the use of free ?

—
Reply to this email directly or view it on GitHub<https://github.com/xbmc/xbmc/pull/8450/files#r46061875>.